core/src/test/java/org/infinispan/conflict/resolution/ConflictResolutionManagerTest.java

```diff
@@ -0,0 +1,188 @@
+import org.infinispan.statetransfer.StateConsumer;
+import org.infinispan.statetransfer.StateRequestCommand;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.topology.CacheTopology;
+import org.infinispan.tx.dld.ControlledRpcManager;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * @author Ryan Emerson
+ * @since 9.0
+ */
+@Test(groups = "functional", testName = "conflict.resolution.ConflictResolutionManagerTest")
+public class ConflictResolutionManagerTest extends MultipleCacheManagersTest {
+
+   private static final int NUMBER_OF_OWNERS = 2;
+   private static final int NUMBER_OF_NODES = NUMBER_OF_OWNERS + 2;
+   private static final int NUMBER_OF_CACHE_ENTRIES = 100;
+   private static final int INCONSISTENT_VALUE_INCREMENT = 10;
+   private static final int NULL_VALUE_FREQUENCY = 20;
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      createClusteredCaches(NUMBER_OF_NODES, getConfigurationBuilder());
+   }
+
+   private ConfigurationBuilder getConfigurationBuilder() {
+      ConfigurationBuilder cacheConfigBuilder = getDefaultClusteredCacheConfig(CacheMode.DIST_SYNC);
+      cacheConfigBuilder.clustering()
+            .hash().numOwners(NUMBER_OF_OWNERS)
+            .stateTransfer().fetchInMemoryState(true);
+      return cacheConfigBuilder;
+   }
+
+   @BeforeMethod
+   public void populateCaches() {
+      AdvancedCache<Integer, String> cache0 = advancedCache(0);
+      IntStream.range(0, NUMBER_OF_CACHE_ENTRIES).forEach(i -> cache0.put(i, "v" + i));
+   }
+
+   @Test(expectedExceptions = IllegalStateException.class,
+         expectedExceptionsMessageRegExp = ".* Unable to retrieve conflicts for key .*")
+   public void testGetAllVersionsDuringStateTransfer() {
+      AdvancedCache cache0 = advancedCache(0);
+      StateConsumer sc = TestingUtil.extractComponent(cache0, StateConsumer.class);
+
+      sc = spy(sc);
+      doReturn(true).when(sc).isStateTransferInProgressForKey(any());
+      TestingUtil.replaceComponent(cache0, StateConsumer.class, sc, true);
+      cache0.getConflictResolutionManager().getAllVersions("testKey");
+   }
+
+   @Test(expectedExceptions = IllegalStateException.class,
+   expectedExceptionsMessageRegExp = ".* Unable to retrieve conflicts as StateTransfer is currently in progress for this cache")
+   public void testGetConflictsDuringStateTransfer() {
+      assertClusterSize("Wrong number of caches.", NUMBER_OF_NODES);
+
+      AdvancedCache cache0 = advancedCache(0);
+      ControlledRpcManager controlledRpcManager = new ControlledRpcManager(TestingUtil.extractComponent(cache0, RpcManager.class));
+      TestingUtil.replaceComponent(cache0, RpcManager.class, controlledRpcManager, true);
+      controlledRpcManager.blockAfter(StateRequestCommand.class);
+
+      StateConsumer sc = TestingUtil.extractComponent(cache0, StateConsumer.class);
```

Comment:
This does not look like really starting state transfer, you just change topology, without real rehash happening.